### DISCUSSION TITLE
Hitoverride behaviour when hitdef have p1stateno or p2stateno

### DISCUSSION THREAD

**Discussion by Kiwi1444** [ikemen-engine/Ikemen-GO#2230] [Feature Request] [answered]
### Describe the bug

Here what I readed on MFG forum:

[_Hitoverride does not play nice with p2stateno in a hitdef. If hitdef uses P2stateno, anything with a hitoverride is completely immune to it (assuming it would override etc etc) If i'm overriding projectiles, and you hit me with a projectile that has p2stateno on it, the projectile will miss exactly as if i had nothitby = SCA active._ ](https://mugenguild.com/forum/msg.2298781)

_The same applies for P1StateNo_

### To Reproduce

1. Add this code to -2:
```
State -2, HitOverride]
type = HitOverride
trigger1 = 1
attr = SCA,AA,AP,AT
stateno = 120
ignorehitpause = 1
```
2. Be hited by anything with p1stateno or p2stateno in hitdef: hepler or player. Attack will be ignored like you invincible

### Expected behavior

Normal hitoverride behaviour

### Engine Version (or source code date)

Nightly 2024-12-17

### Operating system

Windows

**Comment by potsmugen** [comment]
This is shared with Mugen. Elecbyte even listed it under their known issues. We could look into fixing it but I wonder if it'd break any characters. Maybe such a fix would require ikemenversion.

It's also kind of hard to define what should happen in those interactions. Say you have HitOverride against throws and the enemy tries to throw you (has p2stateno). It's clear that scenario ought to miss. But then what if it's an attack with p2stateno like Fast Kung Fu Palm? There'll be cases where it should hit and others where it should miss. Maybe this is why Elecbyte just made them miss while they figured things out.

**Comment by K4thos** [comment]
If it works like Mugen, then it's neither a bug nor a compatibility issue. Transferring to discussion instead.

**Comment by potsmugen** [comment, marked as answer]
We already have a parameter than unhardcodes this behavior: `missOnOverride`.
